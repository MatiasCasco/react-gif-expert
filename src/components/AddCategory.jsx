import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        console.log(target.value);
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        console.log(inputValue);
        if( inputValue.trim().length <=1 ) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return(
        // Se agrega aria-label para que testing-library encuentre el form
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type='text'
                placeholder='Buscar gifs'
                value={ inputValue }
                onChange={onInputChange}
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}