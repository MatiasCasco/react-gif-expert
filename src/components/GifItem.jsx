export const GifItem = ({title, url, id}) => {
        // console.log({title, url});
        return (
            <div className='card'>
                    <img src={url} alt={title}/>
                    <p>{title}</p>
            </div>
        );
}
