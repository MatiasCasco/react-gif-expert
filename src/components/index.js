export * from './GifGrid.jsx';
export * from './AddCategory.jsx';
export * from './GifItem.jsx';