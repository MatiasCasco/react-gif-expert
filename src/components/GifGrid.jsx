import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { GifItem } from './GifItem.jsx';
import { useFetchGifs } from '../hooks/useFetchGifs.js';

export const GifGrid = ({ category }) => {

    const { images, isLoading} = useFetchGifs(category);



    return (
      <>
          <h3>{ category }</h3>
          {isLoading && (<h2>cargando....</h2>)}
          <div className='card-grid'>
              {
                  images.map( image => (
                     <GifItem
                         key={image.id}
                         {...image}
                     />
                  ))
              }
          </div>
      </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}