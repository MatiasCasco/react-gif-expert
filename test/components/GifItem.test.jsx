import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem/>', () => {

    const title = 'Google';
    const url = 'https://www.google.com';

    test('Debe de hacer match con el snapshot', () => {
        const { asFragment, container } = render(<GifItem title={title} url={url} />);
        expect(asFragment()).toMatchSnapshot();
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con la url y alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(src);
        expect(alt).toBe(alt);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
