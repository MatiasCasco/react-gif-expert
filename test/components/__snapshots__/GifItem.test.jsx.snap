// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="Google"
      src="https://www.google.com"
    />
    <p>
      Google
    </p>
  </div>
</DocumentFragment>
`;

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 2`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Google"
      src="https://www.google.com"
    />
    <p>
      Google
    </p>
  </div>
</div>
`;
